src/rpdk/core/contract/resource_client.py

```diff
@@ -242,6 +271,30 @@ def assert_time(start_time, end_time, action):
             "Handler %r timed out." % action
         )
 
+    @staticmethod
+    def get_identifier_list(identifier_path):
+        return [identifier[-1] for identifier in list(identifier_path)]
+
+    @staticmethod
+    def assert_primary_identifier(primary_identifier_path, resource_model):
+        primary_identifiers_list = ResourceClient.get_identifier_list(
+            primary_identifier_path
+        )
+        for primary_identifier in primary_identifiers_list:
+            assert resource_model.get(primary_identifier) is not None
```

Comment:
try using `assert all(resource_model.get(primary_identifier) for primary_identifier in primary_identifiers_list)`

example here:
https://github.com/aws-cloudformation/cloudformation-cli/blob/1194ce3002fee23a4a2917eff6b98c0c65dd55ec/tests/test_test.py#L235-L237

also i think `assert None` throws an error so `assert thing is not None` can be simplified